src/Modules/magic.ts

```diff
@@ -795,6 +850,40 @@ export class MagicModule extends BaseModule {
         }
     }
 
+    // ***************** Voice Casting *******************
+
+    CheckForSpellVoiceCasting(msg: string): void {
+        let spellTargetPair: [SpellDefinition | null, Character | null] | undefined = this.getSpellTargetTupleFromMsg(msg);
+        if (!spellTargetPair || !spellTargetPair[0] || !spellTargetPair[1]) // Skip if no or invalid tuple result
+            return;
+
+        let foundSpell = spellTargetPair[0];
+        let target = spellTargetPair[1];
+        let pairTgt: Character | undefined;
+        if (this.SpellNeedsPair(foundSpell)) {
+            pairTgt = this.PairedCharacterOptions(target)[getRandomInt(this.PairedCharacterOptions(target).length)];
+        }
+        this.CastSpellActual(foundSpell, target, true, pairTgt);
+    }
+
+    getSpellTargetTupleFromMsg(msg: string): [SpellDefinition | null, Character | null] | undefined {
+        let oocParsedString = excludeParentheticalContent(msg); // Don't allow voice casting in OOC chat        
+        for (let s of this.AvailableSpells.filter(s => s.AllowVoiceCast)) { // Only look at spells which allow voice cast
+            if (!s.AllowVoiceCast)
+                continue;
+            let searchPhrase = (!!s.CastingPhrase && s.CastingPhrase.length > 0) ? s.CastingPhrase : s.Name;
+            let re = new RegExp("^" + escapeRegExp(searchPhrase) + " ([\\w\\s]+)(\\b|$|\\s)", "i");
```

Comment:
Like this it would not work when stuttering, this should fix it:
```suggestion
            let re = new RegExp("^(?:\\w-)?" + escapeRegExp(searchPhrase) + " (?:\\w-)?([\\w\\s]+)(\\b|$|\\s)", "i");
```
(I'm bad at regex, there might be a more elegant way to fix it)

Reply:
Is it worth considering that if you are stuttering you're unable to speak the magic phrase?

Reply:
I think yes, because some player spend most of their time stuttering, so it would be too restrictive.
It only exclude the first word stuttering so technically the Magic prhase is still in one piece (ex: "M-Magic Phrase")